### PR TITLE
EOS-18381: Restricted CLI: CLI Setup command to update configuration

### PR DESCRIPTION
**Pull request by avrnv**
# Backend

## Problem Statement
Story Ref (if any): [EOS-18381](https://jts.seagate.com/browse/EOS-18381)
## Unit testing on RPM done
Yes
## Problem Description
The cli_setup utility is required to update the CORTX CLI configuration according to the Mini-provisioner API specification.
## Solution
Implemented the cli_setup utility based on the csm_setup framework.
## Unit Test Cases
- cli_setup config --config \<URL\>

Signed-off-by: Alexander Voronov <alexander.voronov@seagate.com>

### CODE REVIEW

**Review by kupranay — APPROVED** [review]

File: csm/cli/schema/cli_setup.json [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+      "comm": {
+        "type": "direct",
+        "target": "csm.cli.conf.reset",
+        "method": "execute",
+        "class": "Reset",
+        "is_static": false
+      }
+    },
+    {
+      "name": "refresh_config",
```
Comment (pawankumarsrivastava) [inline csm/cli/schema/cli_setup.json:117]: Refresh Config is not required. This has been removed

File: csm/cli/conf/cli_setup.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        cmd_obj.handle_main_parse(subparsers)
+        namespace = parser.parse_args(self._args)
+        sys_module = sys.modules[__name__]
+        for attr in ['command', 'action', 'args']:
+            setattr(sys_module, attr, getattr(namespace, attr))
+            delattr(namespace, attr)
+        return command(action, vars(namespace), args)
+
+    def process(self):
+        ''' Parse args for csm_setup and execute cmd to print output '''
```
Comment (pawankumarsrivastava) [inline csm/cli/conf/cli_setup.py:59]: csm_setup -> cli_setup

File: csm/cli/conf/cli_setup.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        Log.init(service_name = "cli_setup", log_path = const.CSM_SETUP_LOG_DIR,
+                level=const.LOG_LEVEL)
+
+    def _validate(self):
+        ''' Validate setup command '''
+        if len(self._args) < 2:
+            raise Exception('Usage: cli_setup -h')
+
+    def _get_command(self):
+        ''' Parse csm setup command '''
```
Comment (pawankumarsrivastava) [inline csm/cli/conf/cli_setup.py:44]: csm_setup -> cli_setup

File: csm/cli/conf/cli_setup.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import traceback
+import pathlib
+
+class CliSetupCommand:
+    """
+        Provide cli to setup csm. Create user for csm to allow basic
+        permission like log, bundle path.
+    """
+    def __init__(self, argv):
+        ''' Check csm setup command and initialize '''
```
Comment (pawankumarsrivastava) [inline csm/cli/conf/cli_setup.py:29]: csm_setup -> cli_setup

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: cicd/pyinstaller/product_cli.spec [hunk 5]
```diff
@@ -66,6 +66,22 @@ cortxcli = Analysis([cc_path + '/cli/cortxcli.py'],
              cipher=block_cipher,
              noarchive=False)
 
+cli_setup = Analysis([cc_path + '/cli/conf/cli_setup.py'],
+             pathex=['/usr/lib/python3.6/site-packages/'],
```
Comment (udayan-y21) [inline cicd/pyinstaller/product_cli.spec:70]: @avrnv We are planning to remove pyinstaller. Hence `cicd/pyinstaller` directory wont be in picture.
For adding `cli_setup` as binary, please check https://github.com/Seagate/cortx-manager/pull/406/files#diff-2344324d1ead713aeedd540e4d327417154fae8fb76567f006879e42dc498735R282

Just copy cli_setup.py into $DIST/cli/lib

File: csm/cli/schema/cli_setup.json (hunk 1 above)
Comment (avrnv) [reply csm/cli/schema/cli_setup.json]: Removed.

File: csm/cli/conf/cli_setup.py (hunk 2 above)
Comment (avrnv) [reply csm/cli/conf/cli_setup.py]: Done.

File: csm/cli/conf/cli_setup.py (hunk 3 above)
Comment (avrnv) [reply csm/cli/conf/cli_setup.py]: Done.

**Review by pawankumarsrivastava — APPROVED** [review]

**Review by udayan-y21 — APPROVED** [review]

**Review by kupranay — APPROVED** [review]

**Pull request merged**